CMakeLists.txt

```diff
@@ -135,13 +135,19 @@ find_package(Iconv)
 
 pkg_check_modules(LUA REQUIRED IMPORTED_TARGET lua>=5.2)
 pkg_check_modules(POPT REQUIRED IMPORTED_TARGET popt)
-pkg_check_modules(MAGIC REQUIRED IMPORTED_TARGET libmagic)
 pkg_check_modules(READLINE IMPORTED_TARGET readline)
 pkg_check_modules(ZSTD IMPORTED_TARGET libzstd>=1.3.8)
 pkg_check_modules(LIBELF IMPORTED_TARGET libelf)
 pkg_check_modules(LIBDW IMPORTED_TARGET libdw)
 pkg_check_modules(LIBLZMA IMPORTED_TARGET liblzma>=5.2.0)
 
+# file >= 5.39 ships a pkg-config, may move to that later
+find_path(MAGIC_INCLUDE_DIR NAMES magic.h)
+find_library(MAGIC_LIBRARY NAMES magic)
+if (NOT (MAGIC_INCLUDE_DIR AND MAGIC_LIBRARY))
```

Comment:
The danger of using another cmake newbies code as an example :smile: Just add REQUIRED to both find_ calls and you don't need this manual test.

Perhaps we should make this an imported target though, if only for consistency with everything else.

Reply:
Ohh, I should've mentioned this in the commit message actually - cmake 3.12 doesn't know REQUIRED for  these two functions yet :laughing: Only the recent versions do...

... which brings me to the second point - yes, just like with IMA and Lua, let's do this the same way, as an imported target. I'm on it.

Reply:
(In fact, 3.18 was the version where REQUIRED was added)

Reply:
Oh, argh. This actually rings a bell, and may well have been *the* reason for writing it that way in the IMA case. 

I'm tempted to just bump the damn requirement already :laughing:

Reply:
No public release was made with cmake yet, so yup, this might be the right time :smile: